docs/_downloads/433d4b2ff389555700f26fe734f79b9a/plot_group_lasso.py

```diff
@@ -0,0 +1,215 @@
+#
+# License: BSD (3-clause)
+
+import mne
+import os
+import os.path as op
+from mne.parallel import parallel_func
+from mne.datasets import hf_sef
+from matplotlib import pyplot as plt
+
+from groupmne import (compute_group_inverse, prepare_fwds, get_src_reference,
+                      compute_fwd)
+
+##########################################################
+# Download and process MEG data
+# -----------------------------
+#
+# For this example, we use the HF somatosensory dataset [2].
+# We need the raw data to estimate the noise covariance
+# since only average MEG data (and MRI) are provided in "evoked".
+# The data will be downloaded in the same location
+
+
+_ = hf_sef.data_path("raw")
+data_path = hf_sef.data_path("evoked")
+meg_path = data_path + "/MEG/"
+
+data_path = op.expanduser(data_path)
+subjects_dir = data_path + "/subjects/"
+os.environ['SUBJECTS_DIR'] = subjects_dir
+
+raw_name_s = [meg_path + s for s in ["subject_a/sef_right_raw.fif",
+              "subject_b/hf_sef_15min_raw.fif"]]
+
+
+def process_meg(raw_name):
+    """Extract epochs from a raw fif file.
+
+    Parameters
+    ----------
+    raw_name: str.
+        path to the raw fif file.
+
+    Returns
+    -------
+    epochs: Epochs instance
+
+    """
+    raw = mne.io.read_raw_fif(raw_name)
+    events = mne.find_events(raw)
+
+    event_id = dict(hf=1)  # event trigger and conditions
+    tmin = -0.05  # start of each epoch (50ms before the trigger)
+    tmax = 0.3  # end of each epoch (300ms after the trigger)
+    baseline = (None, 0)  # means from the first instant to t = 0
+    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, proj=True,
+                        baseline=baseline)
+    return epochs
+
+
+epochs_s = [process_meg(raw_name) for raw_name in raw_name_s]
+evokeds = [ep.average() for ep in epochs_s]
+
+# compute noise covariance (takes a few minutes)
+noise_covs = []
+for subj, ep in zip(["a", "b"], epochs_s):
+    cov_fname = meg_path + f"subject_{subj}/sef-cov.fif"
+    if os.path.exists(cov_fname):
+        cov = mne.read_cov(cov_fname)
+    else:
+        cov = mne.compute_covariance(ep, tmin=None, tmax=0.)
+        mne.write_cov(cov_fname, cov)
+    noise_covs.append(cov)
+
+
+f, axes = plt.subplots(1, 2, sharey=True)
+for ax, ev, nc, ll in zip(axes.ravel(), evokeds, noise_covs, ["a", "b"]):
+    picks = mne.pick_types(ev.info, meg="grad")
+    ev.plot(picks=picks, axes=ax, noise_cov=nc, show=False)
+    ax.set_title("Subject %s" % ll, fontsize=15)
+plt.show()
+
+#########################################################
+# Source and forward modeling
+# ---------------------------
+# To guarantee an alignment across subjects, we start by
+# computing (or reading if available) the source space of the average
+# subject of freesurfer `fsaverage`
+# If fsaverage is not available, it will be fetched to the data_path
+
+resolution = 4
+spacing = "ico%d" % resolution
+src_ref = get_src_reference(spacing=spacing, subjects_dir=subjects_dir)
```

Comment:
I would not introduce this function. Just use directly MNE public functions. I would delete it.